api/src/main/java/org/eclipse/microprofile/health/HealthCheckResponse.java

```diff
@@ -166,7 +182,7 @@ public Status getStatus() {
     }
 
     public Optional<Map<String, Object>> getData() {
```

Comment:
Deprecate this

Reply:
Add a new method that doesn't return optional, and if the map is empty, return empty map

Reply:
Add also a helper method for individual values as mentioned in the issue

Reply:
Thanks @xstefank 
Here I kept the OP idea, although I based the logic on the `data` as a `Map`, being the existing getter now deprecated.
I kept the new method name suggestion, though, although it's a pity we have to deprecate `getData()` and promote the `getDataMap()` method usage.
Finally, I kept the suggested signature for `getData(String)`, since an `Optional` seems to make sense to me as well in this very case.

Reply:
Yeah, I don't think this can be workaround, we need a different name for the new method. Maybe it would be better just to deprecate this and mention that it will change signature in 5.0. WDYT?

Reply:
Yeah, I don't think this can be workaround, we need a different name for the new method. Maybe it would be better just to deprecate this and mention that it will change signature in 5.0. WDYT?

Reply:
Thanks @xstefank - +1, getting back to the implementation I pushed previously, deprecating the current `getData()` method and a note about the fact it'll change signature in 5.0 makes sense to me